Testing/persistence

juanmbellini opened the pull request:
# Summary

This PR adds testing over the persistence layer, according to the trello card: [https://trello.com/c/HVp5JpUa](https://trello.com/c/HVp5JpUa). Is the continuation of #13.

# Notes

- The trello card indicates that testing must be performed over the web and the persistence layer. This PR only solves the persistence layer part. The web layer testing was performed in #12 .


# Trello Card

https://trello.com/c/HVp5JpUa

juanmbellini committed:
Add testing for BookRepository

- Include a RepositoriesTestHelper class with a method to assert that two Books are the same (which is not done using equality).

juanmbellini committed:
Add testing for the UserRepository

- Test only save and retrieve actions --> missing lazy loading stuff

juanmbellini committed:
Add lazy loading test

- Add tests method (together with private helper method) to check whether a lazy collection is actually lazy loaded
- Also fix some javadocs

juanmbellini committed:
Improve the readability of the testLazyInitialization class

- Extract some lines of code into separate methods
- Add some comments, and remove others

darkBasic666 commented on src/test/java/wolox/training/repositories/RepositoriesTestHelper.java:
```diff
@@ -0,0 +1,202 @@
+     * @param message A message to be displayed in case the assertion fails.
+     */
+    /* package */
+    static void assertSame(final Book expected, final Book actual, final String message) {
```
maybe separete this asserts in classes by models, clean this class.
If you have more models , this class will has alot of lines of code.

juanmbellini replied:
Done. Check 4ba795a

juanmbellini committed:
Extract the assertSame methods to UserAssertions and BookAssertions

- This avoids the helper class to grow when adding more models

juanmbellini committed:
Create an abstract saving test

- Extract saving test logic into an abstract method which is used by each of the repositories test classes when testing the saving operation
